File: planner/app/src/main/java/com/pro3/planner/views/LinedEditText.java
package com.pro3.planner.views;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;
import android.support.v4.content.res.ResourcesCompat;
import android.util.AttributeSet;
import android.widget.EditText;

import com.pro3.planner.R;

public class LinedEditText extends EditText {
    private Rect mRect;
    private Paint mPaint;
    private boolean mEnabled;

    // we need this constructor for LayoutInflater
    public LinedEditText(Context context, AttributeSet attrs) {
        super(context, attrs);

        mRect = new Rect();
        mPaint = new Paint();
        mPaint.setStyle(Paint.Style.FILL_AND_STROKE);
        mPaint.setColor(ResourcesCompat.getColor(getResources(), R.color.primary_text_color, null)); //SET YOUR OWN COLOR HERE
    }

    @Override
    protected void onDraw(Canvas canvas) {
        //int count = getLineCount();

        int height = getHeight();
        int line_height = getLineHeight();

        int count = height / line_height;

        if (getLineCount() > count)
            count = getLineCount();//for long text with scrolling

        Rect r = mRect;
        Paint paint = mPaint;
        int baseline = getLineBounds(0, r);//first line

        for (int i = 0; i < count; i++) {

            canvas.drawLine(r.left, baseline + 1, r.right, baseline + 1, paint);
            baseline += getLineHeight();//next line
        }

        super.onDraw(canvas);
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        try {
            if (!mEnabled) return;
            super.setEnabled(false);
            super.setEnabled(mEnabled);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void setEnabled(boolean enabled) {
        this.mEnabled = enabled;
        super.setEnabled(enabled);
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/elements/note/views/LinedEditText.kt
package com.sunilson.firenote.presentation.elements.note.views

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect
import android.support.v4.content.res.ResourcesCompat
import android.util.AttributeSet
import android.widget.EditText
import com.sunilson.firenote.R

class LinedEditText(context: Context, attrs: AttributeSet) : EditText(context, attrs) {

    private val paint: Paint = Paint()
    private val rect: Rect = Rect()
    private var mEnabled: Boolean = false
        set(value) {
            field = value
            super.setEnabled(value)
        }

    init {
        paint.style = Paint.Style.FILL_AND_STROKE
        paint.color = ResourcesCompat.getColor(resources, R.color.primary_text_color, null)
    }

    override fun onDraw(canvas: Canvas?) {

        //Calculate how many lines the EditText currently has
        var count = height / lineHeight
        if (lineCount > count) count = lineCount

        var baseline = getLineBounds(0, rect)

        //Draw a line for each visible line
        for (i in 0 until count) {
            canvas?.drawLine(rect.left.toFloat(), (baseline + 1).toFloat(), rect.right.toFloat(), (baseline + 1).toFloat(), paint)
            baseline += lineHeight
        }

        super.onDraw(canvas)
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        try {
            if (!isEnabled) return
            super.setEnabled(false)
            super.setEnabled(mEnabled)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}